README.md

```diff
@@ -1,3 +1,162 @@
+- **Target OS**: Ubuntu 20.04+ or Debian 11+
+- **Hardware**: Meets [FOGO's official requirements](https://docs.fogo.io/running-a-node.html)
+- **Network**: Open ports for gossip (8001) and dynamic port range (8901-9000)
+
+## Installation
+
+Install from Ansible Galaxy:
+
+```bash
+ansible-galaxy collection install firstset.fogo_community
+```
+
+Or install from source:
+
+```bash
+git clone https://github.com/Firstset/fogo-community-ansible.git
+cd fogo-community-ansible
+ansible-galaxy collection build
+ansible-galaxy collection install firstset-fogo_community-*.tar.gz
+```
+
+## Quick Start
+
+### Basic Validator Setup
+
+Create a playbook to deploy a FOGO validator:
+
+```yaml
+---
+- name: Deploy FOGO Validator
+  hosts: validator_nodes
+  become: true
+  roles:
+    - firstset.fogo_community.validator_service
+```
+
+Run the playbook:
+
+```bash
+ansible-playbook -i inventory site.yml --ask-become-pass
+```
+
+### Advanced Configuration
+
+Override default settings by setting variables:
+
+```yaml
+---
+- name: Deploy FOGO Validator with Custom Config
+  hosts: validator_nodes
+  become: true
+  vars:
+    validator_client_version: v6.0.0
+    firedancer_gossip_port: 8001
+    accounts_path: "/fast-storage/accounts"
+    bootstrap_disks: true
+  roles:
+    - firstset.fogo_community.validator_service
+```
+
+## Included Roles
+
+### `validator_service`
+
+The main role for deploying and managing FOGO validators. See the [role documentation](roles/validator_service/README.md) for detailed configuration options and examples.
+
+**Key tasks performed:**
+- System dependencies installation
+- Optional disk bootstrapping and formatting
+- CPU performance mode configuration
+- UFW firewall setup
+- Service user creation
+- Firedancer binary compilation
+- Configuration file generation
+- Systemd service setup and activation
+
+## Configuration
+
+All configurable variables are documented in [`roles/validator_service/defaults/main.yml`](roles/validator_service/defaults/main.yml). Key variables include:
+
+| Variable | Default | Description |
+|----------|---------|-------------|
+| `validator_client_version` | `v6.0.0` | FOGO release version to install |
+| `service_user` | `fogo` | System user for the validator service |
+| `firedancer_gossip_port` | `8001` | Port for gossip network communication |
+| `bootstrap_disks` | `false` | Whether to detect and format additional disks |
+| `upgrade_only` | `false` | Only upgrade binaries without full setup |
+
+## Common Usage Patterns
+
+### Initial Deployment
+
+```bash
+# Full setup including disk bootstrapping
+ansible-playbook site.yml -e "bootstrap_disks=true" --ask-become-pass
+```
+
+### Binary Upgrades
+
+```bash
+# Upgrade to new version
+ansible-playbook site.yml -e "upgrade_only=true validator_client_version=v6.1.0" --ask-become-pass
+```
+
+### Configuration Updates
+
+```bash
+# Update configuration without rebuilding
+ansible-playbook site.yml --tags config --ask-become-pass
```

Comment:
Updating configuration (and restarting the service) is a good feature, which has not been implemented yet.